src/smt/smt_engine_state.cpp

```diff
@@ -0,0 +1,302 @@
+/*********************                                                        */
+/*! \file smt_engine_state.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Utility for maintaining the state of the SMT engine.
+ **/
+
+#include "smt/smt_engine_state.h"
+
+#include "options/smt_options.h"
+#include "smt/smt_engine.h"
+
+namespace CVC4 {
+namespace smt {
+
+SmtEngineState::SmtEngineState(SmtEngine& smt)
+    : d_smt(smt),
+      d_context(new context::Context),
+      d_userContext(new context::UserContext),
+      d_pendingPops(0),
+      d_fullyInited(false),
+      d_queryMade(false),
+      d_needPostsolve(false),
+      d_status(),
+      d_expectedStatus(),
+      d_smtMode(SmtMode::START)
+{
+}
+
+void SmtEngineState::notifyExpectedStatus(const std::string& status)
+{
+  Assert(status == "sat" || status == "unsat" || status == "unknown")
```

Comment:
Yes I think so.  Notice this is an `Assert` so it only happens in debug mode.  Essentially this level of debugging makes the SmtEngineState defensive as a standalone utility, in case SmtEngine or other classes have further uses of it.